src/main/java/uk/gov/hmcts/reform/ccd/service/remote/DisposeTasksRemoteOperation.java

```diff
@@ -1,73 +1,70 @@
+import uk.gov.hmcts.reform.ccd.util.SecurityUtil;
 import uk.gov.hmcts.reform.ccd.util.log.TasksDeletionRecordHolder;
 
-import static uk.gov.hmcts.reform.ccd.util.RestConstants.DELETE_TASKS_PATH;
-
 @Service
 @Slf4j
 public class DisposeTasksRemoteOperation implements DisposeRemoteOperation {
 
-    private final ParameterResolver parameterResolver;
-
-    private final Gson gson = new Gson();
+    private final SecurityUtil securityUtil;
 
     private TasksDeletionRecordHolder tasksDeletionRecordHolder;
 
-    private final CcdRestClientBuilder ccdRestClientBuilder;
+    private final TasksClient tasksClient;
 
     @Autowired
-    public DisposeTasksRemoteOperation(final CcdRestClientBuilder ccdRestClientBuilder,
-                                       final ParameterResolver parameterResolver,
+    public DisposeTasksRemoteOperation(final TasksClient tasksClient,
+                                       final SecurityUtil securityUtil,
                                        final TasksDeletionRecordHolder tasksDeletionRecordHolder) {
-        this.ccdRestClientBuilder = ccdRestClientBuilder;
-        this.parameterResolver = parameterResolver;
+        this.tasksClient = tasksClient;
+        this.securityUtil = securityUtil;
         this.tasksDeletionRecordHolder = tasksDeletionRecordHolder;
     }
 
+    @SuppressWarnings("java:S1135")
     @Override
     public void delete(final CaseData caseData) {
         final String caseRef = caseData.getReference().toString();
         try {
             final DeleteTasksRequest tasksDeletePostRequest =
-                    new DeleteTasksRequest(new DeleteCaseTasksAction(caseRef));
+                new DeleteTasksRequest(new DeleteCaseTasksAction(caseRef));
 
-            final String requestDeleteBody = gson.toJson(tasksDeletePostRequest);
 
-            final Response taskDeleteResponse = deleteTasks(requestDeleteBody);
+            final Response taskDeleteResponse = deleteTasks(tasksDeletePostRequest);
 
-            logTasksDisposal(caseRef, taskDeleteResponse);
+            tasksDeletionRecordHolder.setCaseTasksDeletionResults(caseRef, taskDeleteResponse.status());
 
-            if (taskDeleteResponse.getStatus() != 201) {
-                log.error("Error in deleting tasks for case : Case Ref = {} ", caseRef);
-            }
+            if (taskDeleteResponse.status() != 201) {
+                final String errorMessage = String
+                    .format("Unexpected response code %d while deleting tasks for case: %s",
+                            taskDeleteResponse.status(), caseData.getReference()
+                    );
 
+                throw new TasksDeletionException(errorMessage);
+            }
         } catch (final Exception ex) {
+            // TODO: we need to re-throw the exception here once task deletion endpoint is enabled in PROD.
+            // TODO: rethrowing the exception will prevent the case deletion in CCD
```

Comment:
Why not create a task in Jira that would be blocked until "task deletion endpoint is enabled in PROD"?